Lib/test/test_file2k.py

```diff
@@ -652,6 +652,38 @@ def io_func():
             self.f.writelines('')
         self._test_close_open_io(io_func)
 
+    def test_iteration_torture(self):
+        # bpo-31530: Crash when concurrently iterate over a file.
+        with open(self.filename, "wb") as fp:
+            for i in xrange(2**20):
+                fp.write(b"0"*50 + b"\n")
+        with open(self.filename, "rb") as f:
+            def iterate():
+                try:
+                    for l in f:
+                        pass
+                except IOError:
+                    pass
```

Comment:
It's probably a better stress test to continue looping even when you get an exception.

Reply:
This makes the test tooooooo slow. I have killed it after 8 minutes.

Reply:
Decrease the size of the file then?

Reply:
2**12 lines is not enough for provoking a crash, 2**13 to 2**14 lines provoke random crashes, and 2**15 lines is needed for stable crashing. After patching the test with 2**13 lines takes about 40 sec, and 2**15 lines need 3 minutes to finish. This is still too long.